src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -160,23 +166,39 @@ export class RetireJsEngine implements RuleEngine {
 					// Convert the output into RuleResult objects and resolve to that.
 					res(this.processOutput(stderr, invocation.rule));
 				} else {
-					// If RetireJS exits with any other code, then it means something went wrong. We'll reject with stderr
-					// for the ease of upstream error handling.
-					rej(stderr);
+					// If RetireJS exits with any other code, then it means something went wrong. The error could be
+					// contained in either stdout or stderr, so we'll send them both to a method for processing, and
+					// reject the string we receive after prefixing it with an informative header.
+					rej(`RetireJS: ${this.processFailure(stdout, stderr)}`);
 				}
 			});
 		});
 	}
 
+	protected processFailure(stdout: string, stderr: string): string {
+		// Either stdout or stderr could contain the error information, depending on what the error was. We'll try the
+		// same tactics on each one, and hopefully we'll find what we're looking for.
+		for (const o of [stdout, stderr]) {
+			try {
+				// It's possible that we're looking at a valid output JSON whose `errors` property contains the errors we want.
```

Comment:
Is it guaranteed that we'll always get a JSON format? If not, can we log the contents in any case so that we still have some information available to troubleshoot?

Reply:
I spent some time doing things that I knew would cause errors, and given the flags we're providing, the output for a failure will either be a JSON with an `errors` property or a raw string describing the error.
Logging is still a good idea.